polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -2129,21 +2129,21 @@ static bool isSigned(QualType Ty) {
 class BinOpInfo {
 public:
   BinOpInfo(ValueCategory LHS, ValueCategory RHS, QualType Ty,
-            BinaryOperator::Opcode Opcode, const BinaryOperator *Expr)
-      : LHS(LHS), RHS(RHS), Ty(Ty), Opcode(Opcode), Expr(Expr) {}
+            BinaryOperator::Opcode Opcode, const Expr *Expr)
+      : LHS(LHS), RHS(RHS), Ty(Ty), Opcode(Opcode), E(Expr) {}
 
   ValueCategory getLHS() const { return LHS; }
   ValueCategory getRHS() const { return RHS; }
   constexpr QualType getType() const { return Ty; }
   constexpr BinaryOperator::Opcode getOpcode() const { return Opcode; }
-  constexpr const BinaryOperator *getExpr() const { return Expr; }
+  constexpr const Expr *getExpr() const { return E; }
 
 private:
   const ValueCategory LHS;
   const ValueCategory RHS;
   const QualType Ty;                   // Computation Type.
   const BinaryOperator::Opcode Opcode; // Opcode of BinOp to perform
-  const BinaryOperator *Expr;
+  const Expr *E;
```

Comment:
I see the type of `E` is now a generic `Expr`. What was the motivation for not using `BinaryOperator`?

Reply:
To be able to reuse this to perform `-X == 0 - X` (for integers).